test/e2e/far_e2e_test.go

```diff
@@ -130,16 +131,32 @@ var _ = Describe("FAR E2e", func() {
 			makeNodeUnready(selectedNode)
 
 			startTime = time.Now()
-			far := createFAR(nodeName, fenceAgent, testShareParam, testNodeParam)
+			far := createFAR(nodeName, fenceAgent, testShareParam, testNodeParam, remediationStrategy)
 			DeferCleanup(deleteFAR, far)
 		})
-		When("running FAR to reboot two nodes", func() {
+		When("running FAR with ResourceDeletion remediation strategy to reboot two nodes", func() {
+			BeforeEach(func() {
+				remediationStrategy = v1alpha1.ResourceDeletionRemediationStrategy
+			})
+			It("should successfully remediate the first node", func() {
+				checkRemediation(nodeName, nodeBootTimeBefore, pod, remediationStrategy)
+				remediationTimes = append(remediationTimes, time.Since(startTime))
+			})
+			It("should successfully remediate the second node", func() {
+				checkRemediation(nodeName, nodeBootTimeBefore, pod, remediationStrategy)
+				remediationTimes = append(remediationTimes, time.Since(startTime))
+			})
+		})
+		When("running FAR with OutOfServiceTaint remediation strategy to reboot two nodes", func() {
+			BeforeEach(func() {
+				remediationStrategy = v1alpha1.OutOfServiceTaintRemediationStrategy
+			})
 			It("should successfully remediate the first node", func() {
-				checkRemediation(nodeName, nodeBootTimeBefore, pod)
+				checkRemediation(nodeName, nodeBootTimeBefore, pod, remediationStrategy)
 				remediationTimes = append(remediationTimes, time.Since(startTime))
 			})
 			It("should successfully remediate the second node", func() {
-				checkRemediation(nodeName, nodeBootTimeBefore, pod)
+				checkRemediation(nodeName, nodeBootTimeBefore, pod, remediationStrategy)
 				remediationTimes = append(remediationTimes, time.Since(startTime))
 			})
```

Comment:
While creating the e2e tests we had in mind of three worker nodes in the cluster where we [randomly pick one](https://github.com/medik8s/fence-agents-remediation/pull/92/files#diff-b9729ac4458c7637438e3bf558999b35adedcaf4d13af1e3eb752db501fd4997R104) and filter it from the list of nodes for the next e2e test. When there are 2 e2e tests, it is okay, but with 4 there are not enough worker nodes. Therefore the fourth e2e test won't have a node to test in the current code.

Either discard the filtering of nodes or reset the list of nodes that can be remediated after the first two e2e tests or just discard the last e2e test.

Reply:
Thank you for the clear clarification. I understand the current situation.
I will change the e2e tests to discard the filtering of nodes before running the next e2e block.

Reply:
It became complicated when trying to skip the e2e test for OutOfServiceTaint remediation on OCP4.12.
So I chose to discard last e2e test and changed the following patches from the last commit:
https://github.com/medik8s/fence-agents-remediation/pull/92/commits/c3c496d758bbe075a16d0f0b56dfbab594e82fa6 This patch is  a new implementation of the e2e test.
https://github.com/medik8s/fence-agents-remediation/pull/92/commits/ed764b60d660e75d00cb2a258ff975d1d399d0b2 This patch implements a env variable way to skip the e2e test for OutOfServiceTaint remediation.

There are nothing changed for the rest of the patches.